flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnApplicationClusterEntryPoint.java

```diff
@@ -129,16 +129,12 @@ private static PackagedProgramRetriever getPackagedProgramRetriever(
             final Configuration configuration,
             final String[] programArguments,
             @Nullable final String jobClassName)
-            throws IOException {
+            throws FlinkException {
 
         final File userLibDir = YarnEntrypointUtils.getUsrLibDir(configuration).orElse(null);
         final File userApplicationJar = getUserApplicationJar(userLibDir, configuration);
-        final ClassPathPackagedProgramRetriever.Builder retrieverBuilder =
-                ClassPathPackagedProgramRetriever.newBuilder(programArguments)
-                        .setUserLibDirectory(userLibDir)
-                        .setJarFile(userApplicationJar)
-                        .setJobClassName(jobClassName);
-        return retrieverBuilder.build();
+        return PackagedProgramRetrieverImpl.create(
+                userLibDir, userApplicationJar, jobClassName, programArguments, configuration);
```

Comment:
I switched to actual factory methods because we didn't use the builder functionality. `@Nullable` values are passed.  The production code boiled down to two different signatures reflected in the `PackagedProgramRetrieverImpl.create` factory methods.